python/ray/serve/_private/router.py

```diff
@@ -891,6 +891,133 @@ async def assign_replica(
         return replica.send_query(query)
 
 
+
+class MagicReplicaScheduler(ReplicaScheduler):
+    """Chooses a replica MAGICALLY."""
+
+    def __init__(
+        self,
+        event_loop: asyncio.AbstractEventLoop,
+        deployment_id: DeploymentID,
+    ):
+        self._loop = event_loop
+        self._deployment_id = deployment_id
+
+        # replica_id -> info.
+        self._replicas: Dict[str, ReplicaWrapper] = {}
+        # replica_id -> cached queue length.
+        self._replica_queues: Dict[str, int] = {}
+
+        # NOTE(edoakes): Python 3.10 removed the `loop` parameter to `asyncio.Event`.
+        # Now, the `asyncio.Event` will call `get_running_loop` in its constructor to
+        # determine the loop to attach to. This class can be constructed for the handle
+        # from a different loop than it uses for scheduling, so we need to construct it
+        # lazily to avoid an error due to the event being attached to the wrong loop.
+        self._lazily_constructed_replicas_updated_event: Optional[asyncio.Event] = None
```

Comment:
I think we shorten this up to just `_lazy_replicas_updated_event`, since lazy semantic is a pretty common one